### PR TITLE
feat: add first batch of flow Button examples

### PR DESCRIPTION
**Pull request by fhdhsni**
Adds the following examples for Flow Button

- basics
- styles
- error
- size
- tertiary-inline
- success
- contrast
- icons
- images
- disabled

### CODE REVIEW

File: src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.router.Route;
+import com.vaadin.demo.DemoExporter; // hidden-source-line
+
+@Route("button-disabled")
+public class ButtonDisabled extends Div {
+    public ButtonDisabled() {
+        // tag::snippet[]
+        Button primaryButton = new Button("Primary");
+        primaryButton.getElement().setAttribute("disabled", "true");
```
Comment (fhdhsni) [inline src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java:14]: I guess `setEnabled(false)` is more appropriate

File: src/main/java/com/vaadin/demo/component/button/ButtonImages.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.vaadin.demo.DemoExporter; // hidden-source-line
+
+@Route("button-images")
+public class ButtonImages extends Div {
+    public ButtonImages() {
+        StreamResource src = new StreamResource(
+                "vaadin-logo-dark.png",
+                () -> getClass().getResourceAsStream("/images/vaadin-logo-dark.png"));
+        // tag::snippet[]
+        Image img = new Image(src, "Vaadin logo");
```
Comment (sissbruecker) [inline src/main/java/com/vaadin/demo/component/button/ButtonImages.java:17]: Seems there is an easier way without having to create a StreamResource, by just passing the file path to the Image constructor. Can check the corresponding example in the old docs: https://vaadin.com/components/vaadin-button/java-examples

Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonImages.java:17]: stole that from `ScrollerBoth` and `AvatarImage`. Can't find an example of string src path.
[Here](https://vaadin.com/docs/latest/flow/application/resources) it says the path is dependant to deployment method.
`/src/main/wepapp/images/vaadin-logo-dark.png` and `/src/main/resources/META-INF/resources/images/vaadin-logo-dark.png` doesn't seem to work.

File: src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java (hunk 1 above)
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java]: done

File: src/main/java/com/vaadin/demo/component/button/ButtonBasic.java [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
-  public ButtonBasic() {
-    // tag::snippet[]
-    Button button = new Button("Button");
-    Text clickedText = new Text("");
-    button.addClickListener(e -> clickedText.setText("The button was clicked"));
+    public ButtonBasic() {
+        // tag::snippet[]
+        Button button = new Button("Button");
+        Paragraph info = new Paragraph(infoText());
+        button.addClickListener(_e -> {
```
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonBasic.java]: done

File: src/main/java/com/vaadin/demo/component/button/ButtonContrast.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.router.Route;
+import com.vaadin.demo.DemoExporter; // hidden-source-line
+
+@Route("button-contrast")
+public class ButtonContrast extends Div {
+    public ButtonContrast() {
+        // tag::snippet[]
+        Button primaryButton = new Button("Primary");
+        primaryButton.addThemeNames("primary contrast");
```
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonContrast.java]: done

File: src/main/java/com/vaadin/demo/component/button/ButtonContrast.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+        Button secondaryButton = new Button("Secondary");
+        secondaryButton.addThemeNames("secondary contrast");
+
+        Button tertiaryButton = new Button("Tertiary (avoid)");
+        tertiaryButton.addThemeNames("tertiary contrast");
+        // end::snippet[]
+        HorizontalLayout horizontalLayout = new HorizontalLayout(primaryButton, secondaryButton, tertiaryButton);
+        horizontalLayout.getThemeList().add("spacing");
```
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonContrast.java]: done

File: src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java]: done

File: src/main/java/com/vaadin/demo/component/button/ButtonImages.java (hunk 2 above)
Comment (sissbruecker) [reply src/main/java/com/vaadin/demo/component/button/ButtonImages.java:18]: I mixed that up and was assuming that the string overload would create a resource under the hood. I see now that it just puts the string into the image href instead. Yeah not sure if that's possible, the image is not even in the Gatsby application. Let's leave it like this for now.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**